sklearn/preprocessing/_encoders.py

```diff
@@ -774,8 +774,8 @@ def _map_drop_idx_to_infrequent(self, feature_idx, drop_idx):
         if infrequent_indices is not None and drop_idx in infrequent_indices:
             categories = self.categories_[feature_idx]
             raise ValueError(
-                f"Unable to drop category {categories[drop_idx]!r} from feature"
-                f" {feature_idx} because it is infrequent"
+                f"Unable to drop category {categories[drop_idx].tolist()!r} from"
```

Comment:
If I understand the NEP correctly then `repr(np.int64(1))` will now look like `np.int64(1)` but `str(np.int64(1))` will (continue) to be `1`. So I think we can just replace the `!r` with `!s` no?

But then isn't `categories[drop_idx]` an array ... so why does the formatting change?

Reply:
I think for strings, we generally want a quoting which is why we use `!r`

In other words you want:
```
Unable to drop category 'a' from feature 0
```

instead of
```
Unable to drop category a from feature 0
```